components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -393,12 +408,18 @@ void RRTMGPRadiation::run_impl (const double dt) {
   auto d_surf_lw_flux_up = get_field_in("surf_lw_flux_up").get_view<const Real*>();
   // Output fields
   auto d_tmid = get_field_out("T_mid").get_view<Real**>();
-  using SmallPack = ekat::Pack<Real,SCREAM_SMALL_PACK_SIZE>;
-  const int n_lay_w_pack = SCREAM_SMALL_PACK_SIZE*ekat::npack<SmallPack>(m_nlay);
-  view_3d_real d_aero_tau_sw("aero_tau_sw",m_ncol,m_nswbands,n_lay_w_pack);
-  view_3d_real d_aero_ssa_sw("aero_ssa_sw",m_ncol,m_nswbands,n_lay_w_pack);
-  view_3d_real d_aero_g_sw  ("aero_g_sw"  ,m_ncol,m_nswbands,n_lay_w_pack);
-  view_3d_real d_aero_tau_lw("aero_tau_lw",m_ncol,m_nlwbands,n_lay_w_pack);
+
+  // This is annoying to have to have to make extra copies of these, but we
```

Comment:
If I understand correctly, these views are only used to init the yakl arrays. inside the pfor down below, right? If that's correct, I would do
```
if (m_do_aerosol_rad) {
  ...
} else {
   // init yakl arrays to 0 (rather than kokkos views)
}
```
Then, inside the pfor, we could do
```
        // Note that RRTMGP expects ordering (col,lay,bnd) but the FM keeps things in (col,bnd,lay) order
        if (do_aerosol_rad) {
          Kokkos::parallel_for(Kokkos::TeamVectorRange(team, nswbands*nlay), [&] (const int&idx) {
              auto b = idx / nlay;
              auto k = idx % nlay;
              aero_tau_sw(i+1,k+1,b+1) = d_aero_tau_sw(icol,b,k);
              aero_ssa_sw(i+1,k+1,b+1) = d_aero_ssa_sw(icol,b,k);
              aero_g_sw  (i+1,k+1,b+1) = d_aero_g_sw  (icol,b,k);
          }); 
        }
```
ans similarly for the `d_aero_tau_lw` loop.

Reply:
Yeah so the problem I'm wrestling with is, if `m_do_aerosol_rad` is false, then `aero_tau_sw`, etc., won't be provided by any other process, and they are "required" fields when `m_do_aerosol_rad` is true. So we can't put something like 'auto d_aero_tau_sw = get_field_in("aero_tau_sw").get_view<const Real***>();' outside the logical, but we likewise can't do
```
if (m_do_aerosol_rad) {
  auto d_aero_tau_sw = get_field_in("aero_tau_sw").get_view<const Real***>();
}
```
because then when we try to use it later, we get a build fail because `d_aero_tau_sw` was not declared in scope. Am I missing something?

Reply:
Yes, you can do that.

Reply:
Okay, I think I did that.